components/brave_wallet/browser/password_encryptor.cc

```diff
@@ -43,38 +43,25 @@ PasswordEncryptor::DeriveKeyFromPasswordUsingPbkdf2(
   return rv == 1 ? std::move(encryptor) : nullptr;
 }
 
-bool PasswordEncryptor::Encrypt(base::span<const uint8_t> plaintext,
-                                base::span<const uint8_t> nonce,
-                                std::vector<uint8_t>* ciphertext) {
-  if (!ciphertext)
-    return false;
+std::vector<uint8_t> PasswordEncryptor::Encrypt(
+    base::span<const uint8_t> plaintext,
+    base::span<const uint8_t> nonce) {
   crypto::Aead aead(crypto::Aead::AES_256_GCM_SIV);
   aead.Init(key_);
-  *ciphertext = aead.Seal(plaintext, nonce, std::vector<uint8_t>());
-  return true;
+  return aead.Seal(plaintext, nonce, std::vector<uint8_t>());
 }
 
-bool PasswordEncryptor::Decrypt(base::span<const uint8_t> ciphertext,
-                                base::span<const uint8_t> nonce,
-                                std::vector<uint8_t>* plaintext) {
-  if (!plaintext)
-    return false;
+absl::optional<std::vector<uint8_t>> PasswordEncryptor::Decrypt(
+    base::span<const uint8_t> ciphertext,
+    base::span<const uint8_t> nonce) {
   crypto::Aead aead(crypto::Aead::AES_256_GCM_SIV);
```

Comment:
Noting that we're using AES_GCM_SIV which should prevent nonce reuse issues for the sharing of nonces between imported accounts and mnemonics, but it won't protect against re-encryption during migration

Reply:
This is to be addressed in a separate issue